suitcase/tiff/__init__.py

```diff
@@ -1,25 +1,256 @@
+
+        This can alternatively be used to write data to generic buffers rather
+        than creating files on disk. See the documentation for the
+        ``directory`` parameter below.
+
+    Parameters
+    ----------
+    gen : generator
+        xpected to yield ``(name, document)`` pairs
+
+    directory : string, Path or Manager.
+        For basic uses, this should be the path to the output directory given
+        as a string or Path object. Use an empty string ``''`` to place files
+        in the current working directory.
+
+        In advanced applications, this may direct the serialized output to a
+        memory buffer, network socket, or other writable buffer. It should be
+        an instance of ``suitcase.utils.MemoryBufferManager`` and
+        ``suitcase.utils.MultiFileManager`` or any object implementing that
+        inferface. See the suitcase documentation (LINK ONCE WRITTEN) for
+        details.
+
+    file_prefix : str, optional
+        The first part of the filename of the generated output files. This
+        string may include templates as in ``{proposal_id}-{sample_name}-``,
+        which are populated from the RunStart document. The default value is
+        ``{uid}-`` which is guaranteed to be present and unique. A more
+        descriptive value depends on the application and is therefore left to
+        the user.
+
+    **kwargs : kwargs
+        kwargs to be passed to ``tifffile.TiffWriter.save``.
+
+    Returns
+    -------
+    dest : dict
+        dict mapping the 'labels' to lists of file names
+
+    Examples
+    --------
+
+    Generate files with unique-identifer names in the current directory.
+
+    >>> export(gen, '')
+
+    Generate files with more readable metadata in the file names.
+
+    >>> export(gen, '', '{plan_name}-{motors}-')
+
+    Include the experiment's start time formatted as YY-MM-DD_HH-MM.
+
+    >>> export(gen, '', '{time:%%Y-%%m-%%d_%%H:%%M}')
+
+    Place the files in a different directory, such as on a mounted USB stick.
+
+    >>> export(gen, '/path/to/my_usb_stick')
+    """
+    serializer = Serializer(directory, file_prefix, **kwargs)
+    try:
+        for item in gen:
+            serializer(*item)
+    finally:
+        serializer.close()
+
+    return serializer.artifacts
+
+
+class Serializer(event_model.DocumentRouter):
+    """
+    Serialize a stream of documents to TIFF stack(s).
+
+    This creates a file named:
+    ``<directory>/<file_prefix>{stream_name}-{field}.tiff``
+    for every Event stream and field that contains 2D 'image like' data.
+
+    .. warning::
+
+        This process explicitly ignores all data that is not 2D and does not
+        include any metadata in the output file.
+
+
+    .. note::
+
+        This can alternatively be used to write data to generic buffers rather
+        than creating files on disk. See the documentation for the
+        ``directory`` parameter below.
+
+    Parameters
+    ----------
+    gen : generator
```

Comment:
This is a copy/paste error, should only in `export`.

Reply:
fixed